.github/workflows/pre-commit.yaml

```diff
@@ -1,13 +1,14 @@
 ---
 name: Pre-Commit
 on:
-  push:
```

Comment:
Turns out this wasn't the problem, so that's why I left it in. That said, I don't see this as a problem since the types you specified should handle our usecases.